Source/WebCore/platform/MIMETypeRegistry.cpp

```diff
@@ -411,6 +411,7 @@ bool MIMETypeRegistry::isSupportedImageMIMEType(const String& mimeType)
 {
     if (mimeType.isEmpty())
         return false;
+    auto normalizedMIMEType = normalizedImageMIMEType(mimeType);
```

Comment:
I made a big mistake here. The `normalizedImageMIMEType` function was for CURL only. But back two years ago I accidentally reversed the conditional and made it work for non-CURL only; you can see that the comment still claims it’s CURL-only. The effect of this was limited, though, because the function only got called when checking `additionalSupportedImageMIMETypes`.

I am not sure we should be relying on this function that was written just for CURL. Maybe you can help me clean up my mistake, @cdumez? I think we should probably add `"image/jpg"` to `additionalSupportedImageMIMETypeArray` instead.

Reply:
It looks to me that https://github.com/WebKit/WebKit/commit/c650d41f8945032a348d7ffd4e11c9ca2dd75ea7 regressed this by only calling `normalizedImageMIMEType()` for `additionalSupportedImageMIMETypeArray` and no longer for `supportedImageMIMETypeSet`.

Reply:
So my patch seems to restore pre-https://github.com/WebKit/WebKit/commit/c650d41f8945032a348d7ffd4e11c9ca2dd75ea7 behavior. Was there another change that reversed the CURL `#if` check? I couldn't find it.

Reply:
Ah never mind, I got confused by the diff. https://github.com/WebKit/WebKit/commit/c650d41f8945032a348d7ffd4e11c9ca2dd75ea7 did indeed reverse the CURL condition. It also stopped calling normalizedImageMIMEType() for supportedImageMIMETypeSet it seems.

Reply:
I believe that was because `supportedImageMIMETypeSet` is ASCII case insensitive, so even CURL did not need the normalize operation any more for that. Or some reason like that. Anyway, I accidentally made a CURL-only hack into a non-CURL-only hack, so **oops**!

Reply:
I think what happened was that I reversed the sense of the `#if` locally to test that the CURL-only code compiled, but then forgot to switch it back!